server/leafnode.go

```diff
@@ -857,6 +859,12 @@ func (s *Server) initLeafNodeSmap(c *client) {
 	for _, isubj := range ims {
 		c.leaf.smap[isubj]++
 	}
+	// If we have gateways enabled we need to make sure the other side sends us responses
+	// that have been augmented from the original subscription.
+	// TODO(dlc) - Should we lock this down more?
+	if applyGlobalRouting {
+		c.leaf.smap[gwReplyPrefix+"*.>"]++
```

Comment:
Do we possibly have the same issue with `_R_` (with service imports)?

Reply:
_R_ is last hop from server to client IIRC.

Reply:
I could write a new test though.